k-distribution/include/builtin/domains.k

```diff
@@ -755,6 +757,17 @@ module K-IO
   syntax KItem ::= #system ( String ) [function, hook(IO.system), impure]
                  | "#systemResult" "(" Int /* exit code */ "," String /* stdout */ "," String /* stderr */ ")"
 
+  syntax K ::= #appendToFile(String, String) [function]
+             | #lockOpenFileThenAppendAndClose(Int, String) [function]
+             | #appendToOpenFileThenClose(K, Int, String) [function]
+             | #thenUnlockAndClose(K, Int, String) [function]
+             | #thenClose(K, Int) [function]
+  rule #appendToFile(Path:String, Text:String) => #lockOpenFileThenAppendAndClose(#open(Path, "a"), Text)
```

Comment:
Does this open and close the file for each write? Why not collect all coverage during execution and write once at the end? Maybe it's already done; I can't quite read K.

Reply:
it's just for simplicity's sake. if we opened the file once at the beginning and wrote everything at the end, we would either have to implement backend-specific logic for the coverage, which we don't currently have to do, or else we would have to modify the configuration and add a cell containing the file descriptor, which would make the implementation much messier. I am aware this isn't particularly great for performance, but it seems to only be a factor of ~2 slower, which isn't so bad for something that doesn't happen in production.

Reply:
Fair enough; 2X is fine for a coverage tool. When K gets faster, this may need to be revised.